src/Uri.js

```diff
@@ -281,6 +280,13 @@ class Uri {
 					break;
 				default:
 					url += opt_uri;
+
+					var uriScheme = opt_uri.substr(0, opt_uri.indexOf(':'));
+					if (uriScheme &&
+						['.', 'localhost'].every((value) => uriScheme.indexOf(value) === -1)) { // jshint ignore:line
```

Comment:
Hey @kienD,

I did a quick research and I figured out that the `uri scheme` accepts besides letters, digits, plus (+), period (.), or hyphen (-). So, the validation will not recognise `iris.xpc` as a valid protocol.

Apparently, we could simplify to this:
```javascript
var uriScheme = opt_uri.substr(0, opt_uri.indexOf(':'));
if (uriScheme && uriScheme.indexOf('localhost') === -1) {
    url = opt_uri;
}
```

Here are some links:
[https://en.wikipedia.org/wiki/Uniform_Resource_Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)
[https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml](https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml)